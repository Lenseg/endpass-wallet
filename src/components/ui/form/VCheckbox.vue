<template>
  <div class="field">
    <label class="checkbox">
      <input
        type="checkbox"
        v-model="innerValue"
      />
      <slot />
    </label>
  </div>
</template>

<script>
export default {
  name: 'v-checkbox',
  props: {
    value: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      innerValue: this.value,
    };
  },
  watch: {
    value(value) {
      this.innerValue = value;
    },
    innerValue(value) {
      this.$emit('input', value);
    },
  },
};
</script>
