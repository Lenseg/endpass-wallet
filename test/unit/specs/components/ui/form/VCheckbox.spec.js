import { shallow } from '@vue/test-utils';

import VCheckbox from '@/components/ui/form/VCheckbox.vue';

describe('VCheckbox', () => {
  describe('render', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(VCheckbox, {
        slots: {
          default: '<div/>',
        },
      });
    });

    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('v-checkbox');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should render component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('props', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(VCheckbox);
    });

    describe('value', () => {
      it('should change checkbox state', () => {
        wrapper.setProps({ value: true });
        expect(
          wrapper.find('input[type="checkbox"]').element.checked,
        ).toBeTruthy();

        wrapper.setProps({ value: false });
        expect(
          wrapper.find('input[type="checkbox"]').element.checked,
        ).toBeFalsy();
      });
    });
  });
});
