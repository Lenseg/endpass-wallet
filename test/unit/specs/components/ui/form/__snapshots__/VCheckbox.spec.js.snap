// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VCheckbox render should render component 1`] = `
<div
  class="field"
>
  <label
    class="checkbox"
  >
    <input
      type="checkbox"
    />
     
    <div />
  </label>
</div>
`;
